### PR TITLE
[enhance](policy) Support to change table's storage policy if the two policy has same resource

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Before this pr, once the table's storage policy is set and the data has been cooled then it would be impossible for the user to change the table's property. This pr loosened restrictions that the policy could be changed to another one once the both policies share the same resource.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java [hunk 1]
```diff
@@ -189,7 +189,13 @@ private boolean processAlterOlapTable(AlterTableStmt stmt, OlapTable olapTable,
...
             String tableStoragePolicy = olapTable.getStoragePolicy();
-            if (!tableStoragePolicy.isEmpty()) {
+            String currentStoragePolicy = currentAlterOps.getTableStoragePolicy(alterClauses);
+
+            // If the two policy has one same resource, then it's safe for the table to change policy
+            // There would only be the cooldown ttl or cooldown time would be affected
+            if (!Env.getCurrentEnv().getPolicyMgr()
+                    .checkStoragePolicyIfSameResource(tableStoragePolicy, currentStoragePolicy)
+                    && !tableStoragePolicy.isEmpty()) {
                 for (Partition partition : olapTable.getAllPartitions()) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/alter/Alter.java:199]: should use partitions's visible version to check if this table has data or not

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.46 seconds
 stream load tsv:          536 seconds loaded 74807831229 Bytes, about 133 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17161988473 Bytes

**Comment by ByteYue** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 49.15 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17161958697 Bytes

**Comment by ByteYue** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.97 seconds
 stream load tsv:          551 seconds loaded 74807831229 Bytes, about 129 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.9 seconds inserted 10000000 Rows, about 346K ops/s
 storage size: 17162362847 Bytes

**Review by morningman — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**